sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -0,0 +1,132 @@
+                          Continuum - 7-Jan-2011
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    Instrument: Fixed argument = 'kanzelhohe'
+    Wavelength: Fixed argument = astropy.units.quantity.Quantity
+                The physical value of wavelength will belong to any of [5460, 6563, 32768]
+                and units will be Angstroms.
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+    Examples
+    --------
+    >>> from sunpy.net import Fido, attrs as a
+    >>> timerange = a.Time('2015/12/28 00:00:00','2015/12/28 09:03:00')
+    >>> results = Fido.search(timerange, a.Instrument('kanzelhohe'),
+    ...                       a.Wavelength(6563*u.AA))   #doctest: +REMOTE_DATA
+    >>> print(results)  #doctest: +REMOTE_DATA
+    Results from 1 Provider:
+    <BLANKLINE>
+    1 Results from the KanzelhoheClient:
+         Start Time           End Time      ...   Instrument      Wavelength
+           str19               str19        ...     str14           str15
+    ------------------- ------------------- ... -------------- ---------------
+    2015-12-28 00:00:00 2015-12-28 09:03:00 ... Kanzelhohe HA2 6563.0 Angstrom
+    <BLANKLINE>
+    <BLANKLINE>
+    >>> response = Fido.fetch(results)  #doctest: +SKIP
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return list of URLS corresponding to value of input timerange.
+        Parameters
+        ----------
+        timerange: `sunpy.time.TimeRange`
+            time range for which data is to be downloaded.
+        Returns
+        -------
+        urls : list
+            list of URLs corresponding to the requested time range
+        """
+        wave = kwargs['wavelength']
+        wave_float = (wave.min if isinstance(wave, a.Wavelength) else wave.min()).value
+        table = {6563: ['halpha2k/recent', 'halph_fr'],
+                 32768: ['caiia', 'caiik_fi'],
+                 5460: ['phokada', 'bband_fi']}
+        # Checking if value is close enough to a wavelength value.
+        # Converting from one unit to other introduces precision errors.
+        try:
+            # The isclose function in numpy can only check two arrays if they
+            # are close to each other within some precision. Standalone values such
+            # as int, doubles etc can't be checked that way. In order to do that, we put the
+            # two indiviual values in two seperate arrays da and db and then apply
+            # isclose on both those arrays.
+            da = list()
+            da.append(wave_float)
+            for wave_nums in table.keys():
+                db = list()
+                db.append(wave_nums)
+                if np.isclose(da, db, 1e-10, 1e-10):
+                    wave = wave_nums
```

Comment:
We could get the close wavelents with:
```python
wave = list(filter(lambda x: np.isclose(x, wave_float), table.keys()))
```
or running `np.isclose` once:

```python
wave = [wave for wave, close in zip(table.keys(), np.isclose(list(table.keys()), wave_float)) if close]
```
or all in numpy:

```python
wave = np.array(list(table.keys()))[np.isclose(list(table.keys()), wave_float)]
```

Probably in all of these it would be cleaner if the keys from `table` are stored in a variable first.
Then after this an error could be raised if no value found.

Reply:
```python3
wavelengths = table.keys()
wave = list(filter(lambda x: np.isclose(x, wave_float), wavelengths))
if(len(wave) == 0):    #if wave is empty
    raise ValueError("no value is enough closer to allowed wavelengths")

```


Can I have something like this here?

Reply:
That's ok. I will remove the comment `#if wave...` as it's redundant, and maybe add the valid wavelengths to the error message.